editor.planx.uk/src/@planx/components/PlanningConstraints/Public.tsx

```diff
@@ -95,13 +99,45 @@ function Component(props: Props) {
     }
   );
 
-  // XXX handle both/either Digital Land response and custom GIS hookup responses
-  const constraints: Record<string, any> | undefined =
-    data?.constraints || data;
+  // If we have a siteBoundary, additionally fetch classified roads (available nationally) using the buffered siteBoundary,
+  //   skip if we only have an address point because we can't confidently buffer (applicant will see question instead)
+  let classifiedRoadsEndpoint: string = `${process.env.REACT_APP_API_URL}/roads`;
+  if (siteBoundary) {
+    const bufferRadiusInMeters = 10; // avg road width is 5-7m
+    const bufferedSiteBoundary: Feature<Polygon, Properties> = buffer(
+      siteBoundary,
+      bufferRadiusInMeters,
+      { units: "meters", steps: 0 }
+    );
+
+    // Flip the buffered polygon coordinates from [x, y] to [y, x].
+    const flipped: Feature<Polygon, Properties> = flip(bufferedSiteBoundary);
+
+    // Format the flipped coordinates as a space-delimited string for the OGC XML filter param
+    const ogcCoordinates: string = flipped.geometry.coordinates[0].join(" ");
+    classifiedRoadsEndpoint += `?geom=${ogcCoordinates}`;
```

Comment:
made the decision to buffer the siteBoundary on the frontend rather than in the API for a couple of reasons:
- `PlanningConstraints` already handles other geo transformations - eg GeoJSON to WKT for Digital Land, GeoJSON to coordinate rings for ESRI, and now GeoJSON to OGC coordinates for OS Features WFS
- If we ever wanted to expose the buffer number to editors as a configurable input, it'd be straightforward to do with this approach
- keeps all @turf dependences in the editor

Reply:
Thanks for the context here - this is really helpful.

I agree that it's not ideal to have the logic split across domains, I'd be in favour or pulling it together if we can.

The value we're currently getting from including the API seems to be...
 - Hide the OS key
 - Format response in a standard way
 - Facilitate testing
 
The OS key can handled by pointing at the API proxy endpoint which is nice. The other two could be handled (I think!) by encapsulating this in a function/hook/library outside this component.

Happy to go with your judgement here if this feels like it's not worth the effort for what would be gained in consolidating this logic in one place.

Reply:
An alternative could be to wrap this all up in the `/gis/:team` API and make a single request for everything.

Reply:
A frontend library function/hook/library is great food for thought, going to table right here and address comprehensively in a future PR - since other planning constraint logic in this component has the same concern of being split across front & back currently :+1: 

Reasons for forgoing a single request for everying at this stage  (eg different extents of available data, different spatial query methodogolgies, flexibility to call separately if decided, etc) connected to explanation over here: https://github.com/theopensystemslab/planx-new/pull/1467#discussion_r1112754073